components/bl-animatedBox-component/src/index.js

```diff
@@ -0,0 +1,92 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const INFINITE = 'infinite';
+const DEFAULT_ZERO = 0;
+const DEFAULT_ONE = '1s';
+
+export default function AnimatedBoxComponent({ component, elRef, eventHandlers, pods }) {
+  const { style, classList, display, type, duration, delay, loop, autoStart } = component;
+  const { onMouseOver, onMouseOut, onClick, onAnimationStart, onCycleComplete, onAnimationEnd } = eventHandlers;
+
+  const [isAnimating, setIsAnimating] = useState(false);
+  const [hasStarted, setHasStarted] = useState(autoStart);
+  const [isInViewport, setIsInViewport] = useState(null);
+
+  const styles = useStyles(style, duration, delay, loop);
+
+  const handleAnimationEnd = useCallback(() => {
+    setHasStarted(false);
+    setIsAnimating(false);
+    onAnimationEnd();
+  }, []);
+
+  useEffect(() => {
+    const ref = elRef.current;
+
+    if (!ref) {
+      return;
+    }
+
+    const observer = new IntersectionObserver(([entry]) => setIsInViewport(entry.isIntersecting), { threshold: 0.5 });
+
+    observer.observe(ref);
+
+    return () => observer.unobserve(ref);
+  }, [elRef, display]);
+
+  useEffect(() => {
+    if (isInViewport && hasStarted) {
+      setIsAnimating(true);
+    } else {
+      setIsAnimating(false);
+    }
```

Comment:
`setIsAnimating(isInViewport && hasStarted)`

or if there could be present non boolean values
```
const isAnimating = !!(inViewport && hasStarted)
setIsAnimating(iisAnimating)
```